_projects/nested-languages.md

```diff
@@ -24,24 +24,75 @@ tags:
-benefit from the ability to determine the programming language of each part of
-a file. If we have e.g. a PHP file, it is likely that we'd have chunks of html
-inside a few of them. We would like coala to be able to still run meaningful
-analysis on those files.
+Generally, a single source file contains code for exactly one programming language.
+But multiple programming languages can also coexist in a single source file.
+coala would benefit from the ability to determine the programming language of
+each part of a file. If we have e.g. a PHP file, it is likely that we'd have
+chunks of HTML inside a few of them. We would like coala to be able to still
+run meaningful analysis on.
 Other examples include:
 
-* awk inside bash
-* JSON inside Javascript
-* HTML, LaTeX or others inside Jinja2
+* Web templating - languages such as PHP or JSP files mix code into HTML.
+* reStructuredText specify different language for codeblock, means we have a
+block of code which is written in some particular language and it is embedded
+inside reStructuredText syntax.
+
+rstcheckBear of coala checks the syntax of reStructuredText and code blocks
+nested within it.
+
+For example
+
+With bad C++ syntax:
+
+```
+====
+Test
+====
+
+.. code:: cpp
+
+    int main()
+    {
+        return x;
+    }
+
+```
+
+```
+$ rstcheck bad_cpp.rst
+bad_cpp.rst:9: (ERROR/3) (cpp) error: 'x' was not declared in this scope
+
+```
+There are some limited languages for which codeblock can be detected by
+rstcheckBear like (C++, Python, Bash, Doctest, JSON, XML, reStructuredText)
+
+In the above example rstcheckBear implements on reStructuredText and lints
+the codeblock which is written in C++ and is nested within reStructuredText.
+Ideally this project would re-implement the rstcheck's linting of code blocks,
+so the splitting and spawning of linters is done inside coala instead of
+inside rstcheck.
+
+* Markdown also has similar Concept as reStructredText, and the same
+functionality would be available for markdown without a custom bear in coala.
+
+* Another example is IPython notebooks, which is also called Jupyter
+notebooks now a days, which contain e.g. python code.
+
+For example
+
+PEP8NotebookBear of coala detects and fixes PEP8 incompliant code(which is
+written in Python) in Jupyter Notebooks.
+
 
 This project is about enabling coala to deal with those situations and allow
 people to write code analysis similar to how they already do it while being
 applicable to the right locations at the right files.
 
-There are several ways to approach this. The most straightforward would be to identifiy common combinations
-of languages and come up with a clever way to implement support for these combinations individually. A more
-advanced path would be to come up with a more abstract way to define how arbitrary combinations could be supported.
+There are several ways to approach this. The most straightforward would be to
```

Comment:
do not modify lines unless you are improving them.  Only changing the line length isnt helpful.

Reply:
Ya actually that's by mistake

Reply:
undo you mistake